src/graph.rs

```diff
@@ -141,44 +160,49 @@ where
             self.reorder(delta_f, delta_b);
         }
 
-        true
+        Ok(())
     }
 
     /// Forwards depth-first-search
     fn dfs_f<'a>(
         &'a self,
-        n: &'a Node<V>,
+        v: V,
         ub: Order,
         visited: &mut HashSet<V>,
-        delta_f: &mut Vec<&'a Node<V>>,
+        delta_f: &mut Vec<&'a Node<V, E>>,
+        cycle: &mut Vec<(V, V, E)>,
```

Comment:
Doesn't it make more sense to have this function return `Result<(), Vec<(V, V, E)>>` instead?